pylib/mozhginfo/mozhginfo/mozhginfo.py

```diff
@@ -0,0 +1,167 @@
+        # Querying by push ID is perferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the full char representation
+        revisions.append(pushes[push_id]["changesets"][-1])
+
+    return revisions
+
+
+def query_pushid_range(repo_url, start_id, end_id, version=2):
+    """
+    Return an ordered list of revisions (newest push id first).
+
+    repo     - represents the URL to clone a repo
+    start_id - from which pushid to start with (oldest)
+    end_id   - from which pushid to end with (most recent)
+    version  - version of json-pushes to use (see docs)
+    """
+    revisions = []
+    url = "%s?startID=%s&endID=%s&version=%s&tipsonly=1" % (
+        JSON_PUSHES % {"repo_url": repo_url},
+        start_id - 1,  # off by one to compensate for pushlog as it skips start_id
+        end_id,
+        version
+    )
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    pushes = req.json()["pushes"]
+    # pushes.keys() is a list of strings which we need to map to integers
+    # We use reverse in order to return list sorted from newest to oldest push id
+    for push_id in sorted(map(int, pushes.keys()), reverse=True):
+        # Querying by push ID is preferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the 12 char representation
+        revisions.append(pushes[str(push_id)]["changesets"][0])
+
+    return revisions
+
+
+def query_revision_info(repo_url, revision, full=False):
+    """
+    Return a dictionary with meta-data about a push including:
+
+        * changesets
+        * date
+        * user
+    """
+    url = "%s?changeset=%s&tipsonly=1" % (JSON_PUSHES % {"repo_url": repo_url}, revision)
+    if full:
+        url += "&full=1"
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    data = req.json()
+    assert len(data) == 1, "We should only have information about one push"
+    push_id, push_info = data.popitem()
+    push_info["pushid"] = push_id
+    if not full:
+        LOG.debug("Push info: %s" % str(push_info))
+    else:
+        LOG.debug("Requesting the info with full=1 can yield too much unnecessary output "
+                  "to debug anything properly")
+    return push_info
+
+
+def query_full_revision_info(repo_url, revision):
+    """
+    Return the full 40 chars revision number which correspond to the 12 chars
+    revision number we given, and throw an error out if found collision in there.
+
+    :param repo_url: The repo url of a repository e.g. mozilla-inbound
+    :param revision: push revision (40 chars)
+    :type revision: str
+    :return:
+    """
+    try:
+        revision = query_revision_info(repo_url, revision)['changesets'][0]
+    except:
+        raise PushlogAuthError('Unable to retrieve pushlog info. '
+                               'Please check repo_url and revision specified.')
```

Comment:
Please also show what revision and repo_url caused problems on your message. Note that usually the users seeing this message didn't pass the program any repo_url.